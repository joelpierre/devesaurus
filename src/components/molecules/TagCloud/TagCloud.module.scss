@import "~sass/core";

.tag-cloud {
  display: flex;
}
