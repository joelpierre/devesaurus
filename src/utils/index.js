export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
