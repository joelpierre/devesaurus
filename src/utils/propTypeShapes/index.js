import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
