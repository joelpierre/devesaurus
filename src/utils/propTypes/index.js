import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
