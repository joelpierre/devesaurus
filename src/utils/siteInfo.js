import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
