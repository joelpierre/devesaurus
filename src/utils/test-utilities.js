import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
