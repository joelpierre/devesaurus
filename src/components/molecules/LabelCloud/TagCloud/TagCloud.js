import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from '../LabelCloud.module.scss';
import Label from '../../../atoms/Label/Label';
import { mapTaxonomyIcon, mapTaxonomyTheme } from '../../../../utils';

export const PureTagCloudList = ({ allWordpressTag }) => (
  <>
    {allWordpressTag.edges.map(({ node }) => (
      <li key={node.id} className={classNames(styles['label-cloud__item'])}>
        <Label
          classes={styles['label-cloud__label']}
          link={'/tag/' + node.slug}
          theme={mapTaxonomyTheme(node.slug)}
        >
          <FontAwesomeIcon
            icon={mapTaxonomyIcon(node.slug)}
            className={styles['label-cloud__icon']}
          />
          {node.name}
        </Label>
      </li>
    ))}
  </>
);

const TagCloud = ({ classes }) => (
  <ul
    data-test="component-label-cloud"
    className={classNames(styles['label-cloud'], classes)}
  >
    <StaticQuery
      data-test="component-label-cloud-query"
      query={query}
      render={props => <PureTagCloudList {...props} />}
    />
  </ul>
);

TagCloud.defaultProps = {
  classes: undefined,
};

TagCloud.propTypes = {
  classes: PropTypes.string,
};

export default TagCloud;

const query = graphql`
  {
    allWordpressTag {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;
