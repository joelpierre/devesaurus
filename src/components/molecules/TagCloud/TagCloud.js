import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import styles from './TagCloud.module.scss';
import Label from '../../atoms/Label/Label';
import { mapTaxonomyIcon, mapTaxonomyTheme } from '../../../utils';

export const getCloudQuery = taxonomy => {
  let query;

  switch (taxonomy) {
    case 'post_tag':
      query = tagQuery;
      return query;

    case 'category':
      query = categoryQuery;
      return query;

    case 'word_tag':
      query = wordTagQuery;
      return query;

    case 'word_category':
      query = wordCategoryQuery;
      return query;

    default:
      query = wordTagQuery;
      return query;
  }
};

export const PureTagCloud = ({ classes, taxonomy, ...props }) => {
  let obj;
  let link;

  switch (taxonomy) {
    case 'post_tag':
      obj = props.allWordpressTag;
      link = `/tag/`;
      break;

    case 'category':
      obj = props.allWordpressCategory;
      link = `/category/`;
      break;

    case 'word_tag':
      obj = props.allWordpressWpWordTag;
      link = `/word-tag/`;
      break;

    case 'word_category':
      obj = props.allWordpressWpWordCategory;
      link = `/word-category/`;
      break;

    default:
      obj = props.allWordpressWpWordTag;
      link = `/word-tag/`;
  }

  return (
    <ul
      data-test="component-tag-cloud"
      className={classNames(styles['tag-cloud'], classes)}
    >
      {obj &&
        obj.edges.map(({ node }) => (
          <li key={node.id} className={classNames(styles['tag-cloud__item'])}>
            <Label
              classes={styles['tag-cloud__label']}
              link={link + node.slug}
              theme={mapTaxonomyTheme(node.slug)}
            >
              <FontAwesomeIcon
                icon={mapTaxonomyIcon(node.slug)}
                className={styles['tag-cloud__icon']}
              />
              {node.name}
            </Label>
          </li>
        ))}
    </ul>
  );
};

const TagCloud = ({ classes, taxonomy }) => {
  return (
    <StaticQuery
      data-test="component-tag-cloud-query"
      query={getCloudQuery(taxonomy)}
      render={props => (
        <PureTagCloud {...props} taxonomy={taxonomy} classes={classes} />
      )}
    />
  );
};

TagCloud.defaultProps = {
  classes: undefined,
  taxonomy: 'word_tag',
};

TagCloud.propTypes = {
  classes: PropTypes.string,
  taxonomy: PropTypes.oneOf(['category', 'tag', 'word_tag', 'word_category']),
};

export default TagCloud;

const wordTagQuery = graphql`
  {
    allWordpressWpWordTag {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;

const tagQuery = graphql`
  {
    allWordpressTag {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;

const categoryQuery = graphql`
  {
    allWordpressCategory {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;

const wordCategoryQuery = graphql`
  {
    allWordpressWpWordCategory {
      edges {
        node {
          id
          name
          slug
          taxonomy
        }
      }
    }
  }
`;
