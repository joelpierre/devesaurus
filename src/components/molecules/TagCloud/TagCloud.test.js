import React from 'react';
import { shallow } from 'enzyme';

import TagCloud from './TagCloud';
import { checkProps, findByTestAttr, matchSnapshot } from '../../../utils/test';

const defaultProps = {};

/**
 * Factory function to create a ShallowWrapper for the TagCloud component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<TagCloud {...setupProps} />);
};

describe('<TagCloud/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('renders the TagCloud Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-tag-cloud');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(TagCloud, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });
});
