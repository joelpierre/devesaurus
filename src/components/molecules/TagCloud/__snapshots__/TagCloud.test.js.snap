// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PureTagCloud/> should match snapshot 1`] = `
<ul
  className="tag-cloud"
  data-test="component-tag-cloud"
/>
`;
